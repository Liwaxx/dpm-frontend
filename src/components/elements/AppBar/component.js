import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import InputAdornment from '@material-ui/core/InputAdornment';
import TextField from '@material-ui/core/TextField';
import Toolbar from '@material-ui/core/Toolbar';
import PropTypes from 'prop-types';
import Date from '../Date';
import Select from '../Select';
import { IMAGES } from '../../../configs';
// import InputBase from '@material-ui/core/InputBase';
// import SearchIcon from '@material-ui/icons/Search';
// import Search from '../Search';


export default class Component extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ value: event.target.value });
  }

  handleSubmit(event) {
    alert('A search was submitted: ' + this.state.value);
    event.preventDefault();
  }

  _renderSearch() {
    const { classes } = this.props;
    const { value } = this.state;
    const endAdornment = (
      <InputAdornment position="start">
        <IconButton className={classes.submit} onSubmit={this.handleSubmit}>
          <img src={IMAGES.SEARCH} className={classes.searchbutton} />
        </IconButton>
      </InputAdornment>
    );
    const InputProps = {
      disableUnderline: true,
      endAdornment
    };

    return (
      <TextField
        className={classes.bar}
        type="text"
        value={value}
        onChange={this.handleChange}
        placeholder="Cari"
        InputProps={InputProps}
      />
    );
  }

  render() {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <AppBar position="static" className={classes.paper}>
          <Toolbar className={classes.toolbar} disableGutters>
            <Date />
            <div className={classes.grow}>
              <div className={classes.select}>
                <Select />
              </div>
              {this._renderSearch()}
            </div>
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object,
};
