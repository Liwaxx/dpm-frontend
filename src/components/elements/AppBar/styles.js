const styles = ({
  root: {
    width: '100%',
    border: '#F5F7FA',
  },
  paper: {
    boxShadow: 'none',
  },
  grow: {
    display: 'flex',
  },
  toolbar: {
    backgroundColor: '#f2f2f2',
    justifyContent: 'space-between',
  },
  select: {
    marginRight: 10
  },
  bar: {
    backgroundColor: '#ffffff',
    borderRadius: '2px',
    height: '32px',
    padding: '8px',
  },
  submit: {
    height: '24px',
    width: '24px',
  },
  searchbutton: {
    width: 16,
  }
});

export default styles;
