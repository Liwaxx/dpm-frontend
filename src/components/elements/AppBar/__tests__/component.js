import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import AppBar from '../component';

jest.mock('../../Date');
jest.mock('../../Select');
jest.mock('../../../../configs');

describe('AppBar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<AppBar classes={{}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('responds to value change', () => {
    const value = 'search';
    const wrapper = mount(<AppBar classes={{}}/>);
    const input = wrapper.find('input');
    
    input.simulate('change', { target: { value: value } });
    expect(wrapper.state().value).toEqual(value);
  });

});