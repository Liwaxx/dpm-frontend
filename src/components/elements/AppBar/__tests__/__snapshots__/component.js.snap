// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders correctly 1`] = `
<div>
  <header
    className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
  >
    <div
      className="MuiToolbar-root-36 MuiToolbar-regular-38"
    >
      <div>
        <div />
        <div
          className="MuiFormControl-root-40"
        >
          <div
            className="MuiInput-root-44 MuiInput-formControl-45"
          >
            <input
              aria-invalid={false}
              className="MuiInput-input-52"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Cari"
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-57 MuiInputAdornment-positionStart-58"
            >
              <button
                className="MuiButtonBase-root-66 MuiIconButton-root-60"
                disabled={false}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onSubmit={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-65"
                >
                  <img
                    src="/assets/search.png"
                  />
                </span>
                <span
                  className="MuiTouchRipple-root-69"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
